### PR TITLE
Backport PR#281 to SL21c

### PR DESCRIPTION
**Pull request by zhux97**
Backport TpcRS and gating grid code from PR #281 to SL21c. The relevant dependencies in DEV are back-ported as well.

### CODE REVIEW

File: StRoot/StMcEventMaker/StMcEventMaker.cxx [hunk 1]
```diff
@@ -621,7 +625,7 @@ Int_t StMcEventMaker::Make()
 	    mtdHitTable = g2t_mtd_hitTablePointer->GetTable();
 	else
 	    if (Debug()) cerr << "Table g2t_mtd_hit Not found in Dataset " << geantDstI.Pwd()->GetName() << endl;
-
+#if 0
```
Comment (starsdong) [inline StRoot/StMcEventMaker/StMcEventMaker.cxx]: Similar to the discussion in PR#281, Xianglei, does this need to be changed to #if 1 or remove completely? Let us resolve this soon. Thanks

Comment (zhux97) [reply StRoot/StMcEventMaker/StMcEventMaker.cxx]: Xin, it is fine for me to keep it as it is.

Comment (starsdong) [reply StRoot/StMcEventMaker/StMcEventMaker.cxx]: Xianglei, I don't quite understand. With "#if 0", we will be skipping all the BEMC, FCS hits? I would suggest to remove it then?

Comment (zhux97) [reply StRoot/StMcEventMaker/StMcEventMaker.cxx]: oh, I did not realize this, please go ahead to remove it then...

**Comment by klendathu2k** [comment]
@zhux97 @genevb @yfisyak  This back port to SL21c looks reasonable to me...  Codes compile... I will plan to merge this later this afternoon.

**Pull request merged**